### PR TITLE
Add in crewjam/saml dependencies for SAML IdP feature.

### PR DESCRIPTION
**Pull request by mdwn**
The crewjam/saml dependency has been added for the SAML IdP feature.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -214,6 +214,8 @@ require (
 	github.com/chai2010/gettext-go v1.0.2 // indirect
 	github.com/coreos/go-systemd/v22 v22.3.3-0.20220203105225-a9a7ef127534 // indirect
 	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f // indirect
+	github.com/crewjam/httperr v0.2.0 // indirect
+	github.com/crewjam/saml v0.4.12
```
Comment (codingllama) [inline go.mod:218]: Mind moving this out of the "indirects" block and into the big require at the top?

File: go.mod [hunk 2]
```diff
@@ -370,6 +374,7 @@ replace (
 	github.com/julienschmidt/httprouter => github.com/gravitational/httprouter v1.3.1-0.20220408074523-c876c5e705a5
 	github.com/keys-pub/go-libfido2 => github.com/gravitational/go-libfido2 v1.5.3-0.20220630200200-45a8c53e4500
 	github.com/pkg/sftp => github.com/gravitational/sftp v1.13.6-0.20220927202521-0e74d42f8055
+	github.com/russellhaering/goxmldsig => github.com/russellhaering/goxmldsig v1.2.0
```
Comment (codingllama) [inline go.mod:377]: Self-replace here too.

File: go.mod [hunk 3]
```diff
@@ -361,6 +364,7 @@ require (
 // Update also `ignore` in .github/dependabot.yml.
 replace (
 	github.com/coreos/go-oidc => github.com/gravitational/go-oidc v0.0.6
+	github.com/crewjam/saml => github.com/crewjam/saml v0.4.12
```
[the remark this answers is not in the document]
Comment (mdwn) [reply go.mod:367]: Hrm, I think I was convinced I needed to do this from when I was prototyping this and trying to figure out the versioning wonkiness (crewjam has several submodules), but it looks like I'm wrong. Will remove.

**Review by codingllama — APPROVED** [review]
I noticed that `github.com/crewjam/httperr` got dropped on https://github.com/gravitational/teleport/pull/21092/commits/b73427546c22be3d3551027b566def0f5c73e80c (it was an indirect before), just wanted to check if that's intentional.

**Review by rosstimothy — COMMENTED** [review]
I also don't think we want to add things to `go.mod` if they aren't imported. Running `go mod tidy` on this branch results in the following:

```diff
diff --git a/go.mod b/go.mod
index b2b3afc4d8..c403f4d27a 100644
--- a/go.mod
+++ b/go.mod
@@ -36,7 +36,6 @@ require (
        github.com/coreos/go-oidc v2.1.0+incompatible // replaced
        github.com/coreos/go-semver v0.3.1
        github.com/creack/pty v1.1.18
-       github.com/crewjam/saml v0.4.12
        github.com/datastax/go-cassandra-native-protocol v0.0.0-20220706104457-5e8aad05cf90
        github.com/denisenkom/go-mssqldb v0.11.0 // replaced
        github.com/dustin/go-humanize v1.0.1
@@ -334,7 +333,6 @@ require (
        github.com/xlab/treeprint v1.1.0 // indirect
        github.com/youmark/pkcs8 v0.0.0-20181117223130-1be2e3e5546d // indirect
        github.com/yuin/gopher-lua v0.0.0-20220504180219-658193537a64 // indirect
-       github.com/zenazn/goji v1.0.1 // indirect
        go.etcd.io/etcd/client/pkg/v3 v3.5.7 // indirect
        go.opencensus.io v0.24.0 // indirect
        go.opentelemetry.io/otel/exporters/otlp/internal/retry v1.12.0 // indirect
diff --git a/go.sum b/go.sum
index f997d098c2..974647a002 100644
--- a/go.sum
+++ b/go.sum
@@ -300,9 +300,6 @@ github.com/creack/pty v1.1.7/go.mod h1:lj5s0c3V2DBrqTV7llrYr5NG6My20zk30Fl46Y7Do
 github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
 github.com/creack/pty v1.1.18 h1:n56/Zwd5o6whRC5PMGretI4IdRLlmBXYNjScPaBgsbY=
 github.com/creack/pty v1.1.18/go.mod h1:MOBLtS5ELjhRRrroQr9kyvTxUAFNvYEK993ew/Vr4O4=
-github.com/crewjam/httperr v0.2.0/go.mod h1:Jlz+Sg/XqBQhyMjdDiC+GNNRzZTD7x39Gu3pglZ5oH4=
-github.com/crewjam/saml v0.4.12 h1:66Gsd+9iA/8ZGl8W+7DDTlJGWe3RneBFo+Uu/gvlB0w=
-github.com/crewjam/saml v0.4.12/go.mod h1:igEejV+fihTIlHXYP8zOec3V5A8y3lws5bQBFsTm4gA=
 github.com/cznic/golex v0.0.0-20181122101858-9c343928389c/go.mod h1:+bmmJDNmKlhWNG+gwWCkaBoTy39Fs+bzRxVBzoTQbIc=
 github.com/cznic/mathutil v0.0.0-20181122101859-297441e03548/go.mod h1:e6NPNENfs9mPDVNRekM7lKScauxd5kXTr1Mfyig6TDM=
 github.com/cznic/parser v0.0.0-20160622100904-31edd927e5b1/go.mod h1:2B43mz36vGZNZEwkWi8ayRSSUXLfjL8OkbzwW4NcPMM=
@@ -316,7 +313,6 @@ github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/daviddengcn/go-colortext v1.0.0 h1:ANqDyC0ys6qCSvuEK7l3g5RaehL/Xck9EX8ATG8oKsE=
 github.com/daviddengcn/go-colortext v1.0.0/go.mod h1:zDqEI5NVUop5QPpVJUxE9UO10hRnmkD5G4Pmri9+m4c=
-github.com/dchest/uniuri v1.2.0/go.mod h1:fSzm4SLHzNZvWLvWJew423PhAzkpNQYq+uNLq4kxhkY=
 github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
 github.com/dgryski/go-rendezvous v0.0.0-20200823014737-9f7001d12a5f h1:lO4WD4F/rVNCu3HqELle0jiPLLBs70cWOduZpkS1E78=
 github.com/dgryski/go-rendezvous v0.0.0-20200823014737-9f7001d12a5f/go.mod h1:cuUVRXasLTGF7a8hSLbxyZXjz+1KgoB3wDUb6vlszIc=
@@ -1166,8 +1162,6 @@ github.com/yuin/goldmark v1.4.13/go.mod h1:6yULJ656Px+3vBD8DxQVa3kxgyrAnzto9xy5t
 github.com/yuin/gopher-lua v0.0.0-20220504180219-658193537a64 h1:5mLPGnFdSsevFRFc9q3yYbBkB6tsm4aCwwQV/j1JQAQ=
 github.com/yuin/gopher-lua v0.0.0-20220504180219-658193537a64/go.mod h1:GBR0iDaNXjAgGg9zfCvksxSRnQx76gclCIb7kdAd1Pw=
 github.com/zenazn/goji v0.9.0/go.mod h1:7S9M489iMyHBNxwZnk9/EHS098H4/F6TATF2mIxtB1Q=
-github.com/zenazn/goji v1.0.1 h1:4lbD8Mx2h7IvloP7r2C0D6ltZP6Ufip8Hn0wmSK5LR8=
-github.com/zenazn/goji v1.0.1/go.mod h1:7S9M489iMyHBNxwZnk9/EHS098H4/F6TATF2mIxtB1Q=
 go.etcd.io/bbolt v1.3.2/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
 go.etcd.io/bbolt v1.3.3/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
 go.etcd.io/etcd v0.0.0-20191023171146-3cf2f69b5738/go.mod h1:dnLIgRNXwCJa5e+c6mIZCrds/GIG4ncV9HhK5PX7jPg=
@@ -1789,8 +1783,6 @@ gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b/go.mod h1:K4uyk7z7BCEPqu6E+C
 gopkg.in/yaml.v3 v3.0.0/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
 gopkg.in/yaml.v3 v3.0.1 h1:fxVm/GzAzEWqLHuvctI91KS9hhNmmWOoWu0XTYJS7CA=
```

**Comment by mdwn** [comment]
> I noticed that `github.com/crewjam/httperr` got dropped on [b734275](https://github.com/gravitational/teleport/commit/b73427546c22be3d3551027b566def0f5c73e80c) (it was an indirect before), just wanted to check if that's intentional.

Yeah, looks like it's no longer needed.

> I also don't think we want to add things to `go.mod` if they aren't imported. Running `go mod tidy` on this branch results in the following:

So this will end up being used by the enterprise SAML IdP feature, which I've got in a separate branch at the moment -- how do we get around this sort of thing in the meanwhile?

**Review by rosstimothy — APPROVED** [review]

**Comment by codingllama** [comment]
> So this will end up being used by the enterprise SAML IdP feature, which I've got in a separate branch at the moment -- how do we get around this sort of thing in the meanwhile?

Ideally the go.mod changes land together with whatever imports the new modules, but if this is for Enterprise then you could add the dependency to https://github.com/gravitational/teleport/blob/master/e_imports.go - just try to clean it up if it's not needed anymore.

**Pull request merged**